# Chapter 17
# Bulit in error
# Exception how to handle them
# raise your own error, debugging,etc

# before move forward remove  this """ """
"""
# Syntax Error
def func:
pass

# IndentationError: unexpected indent
def func():
pass

# name error
print(name) # printing name without assigning

# type error
print(5+"str")

# index error
l = [1,2,3]
print(l[5])

# values error
str = "abc"
print(int(str))

# Attribute error
l = [1,2,3]
l.push(5) # there is no push function in list 

# key error
d = {"name":"Darshan"}
print(d['age'])
"""